keys/generate-key-pair.sh

```diff
@@ -54,39 +55,73 @@ function allKeys() {
   for i in "rsa" "ecdsa256" "ecdsa384"; do
     keyType=$i
     genKey
-    wait
   done
 }
 
 #This function will create a private key that is needed to create a private keystore. Encryption keyType passed will decide which command to run for private key creation
 function genKey() {
+  local privateKey=""${keyType}"private-key.pem"
+  local keyCert=""${keyType}"Cert.crt"
   #Check if the folder is already created for the keyType (In case of multiple runs)
   mkdir -p "${keyType}"Key && pushd "${keyType}"Key >/dev/null || return
   #Generate a private RSA key.
   if [[ $keyType == "rsa" ]]; then
-    echo -e "Generating a "${keyType}" private key."
-    openssl genrsa -out "${keyType}"private-key.pem 2048 >/var/tmp/out 2>&1
-    chk $? 'Generating a rsa private key.'
+    if [ "$(uname)" == "Darwin" ]; then
+      echo "Using macOS, will generate private key and certificate simultaneously."
+      gencmd="openssl req -x509 -nodes -days 3650 -newkey rsa:2048 -keyout "$privateKey" -out "$keyCert""
+    else
+      echo -e "Generating an "${keyType}" private key."
+      openssl genrsa -out "${keyType}"private-key.pem 2048 >/dev/null 2>&1
+      chk $? 'Generating a rsa private key.'
+    fi
     keyCertGenerator
   #Generate a private ecdsa (256 or 384) key.
   elif [[ $keyType == "ecdsa256" ]] || [[ $keyType == "ecdsa384" ]]; then
     echo -e "Generating an "${keyType}" private key."
     local var2=$(echo $keyType | cut -f2 -da)
-    openssl ecparam -genkey -name secp"${var2}"r1 -out "${keyType}"private-key.pem >/dev/null 2>&1
-    chk $? 'Generating an ecdsa private key.'
+    if [ "$(uname)" == "Darwin" ]; then
+      echo "Using macOS, will generate private key and certificate simultaneously."
+    else
+      openssl ecparam -genkey -name secp"${var2}"r1 -out "${keyType}"private-key.pem >/dev/null 2>&1
+      chk $? 'Generating an ecdsa private key.'
+    fi
     keyCertGenerator
   fi
 }
 
-#This function will create a keyCertificate that is needed to create a corresponding public key, as well as a private keystore.
 function keyCertGenerator() {
   if [[ -f "${keyType}"private-key.pem ]]; then
-    local privateKey=""${keyType}"private-key.pem"
-    local keyCert=""${keyType}"Cert.crt"
-    echo -e ""${keyType}" private key creation: Successful"
-    #Generate a self-signed certificate from the private key file.
-    #You should have these environment variables set. If they aren't you will be prompted to enter values.
-    echo -e "Generating a corresponding certificate."
+    if [ "$(uname)" == "Darwin" ]; then
+      rm "${keyType}"private-key.pem
+    else
+      echo -e ""${keyType}" private key creation: Successful"
```

Comment:
I've mentioned this before: it is easier to read/less confusing if you put only 1 set of double quotes around the whole string. Variables inside the double quoted string will still get evaluated. This comment applies to the whole script, because you have many occurrences of this.

Also, why use `-e` if there are no newlines or other backslashed chars you want honored? It is unnecessary.